src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -626,6 +628,41 @@ export class HaConfigDevicePage extends LitElement {
       updateEntry: async (updates) => {
         const oldDeviceName = device.name_by_user || device.name;
         const newDeviceName = updates.name_by_user;
+        const disabled =
+          updates.disabled_by === "user" && device.disabled_by !== "user";
+
+        if (disabled) {
+          for (const cnfg_entry of device.config_entries) {
+            if (
+              !this.devices.some(
+                (dvc) =>
+                  dvc.id !== device.id &&
+                  dvc.config_entries.includes(cnfg_entry)
```

Comment:
We could potentially do like this, to suggest disabling a config entry with multiple devices when the last device is being disabled:

```suggestion
                  dvc.config_entries.includes(cnfg_entry) &&
                  dvc.disabled_by !== "user"
```